pkg/agent/multicluster/streched_networkpolicy_controller.go

```diff
@@ -0,0 +1,375 @@
+
+	klog.InfoS("Starting MultiCluster Controller", "controllerName", stretchedNetworkPolicyControllerName)
+	defer klog.InfoS("Shutting down MultiCluster Controller", "controllerName", stretchedNetworkPolicyControllerName)
+	cacheSyncs := []cache.InformerSynced{s.podListerSynced, s.namespaceListerSynced, s.LabelIdentityListerSynced}
+	if !cache.WaitForNamedCacheSync(stretchedNetworkPolicyControllerName, stopCh, cacheSyncs...) {
+		return
+	}
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(s.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (s *StretchedNetworkPolicyController) initLabelIDCache() {
+	labelIdentities, err := s.labelIdentityLister.List(labels.Everything())
+	if err == nil {
+		for _, labelIdentity := range labelIdentities {
+			s.labelIdentityCache[labelIdentity.Spec.Label] = labelIdentity.Spec.ID
+		}
+	}
+}
+
+func (s *StretchedNetworkPolicyController) enqueuePod(obj interface{}) {
+	pod, _ := obj.(*v1.Pod)
+	s.queue.Add(pod)
+}
+
+// worker is a long-running function that will continually call the processNextWorkItem
+// function in order to read and process a message on the workqueue.
+func (s *StretchedNetworkPolicyController) worker() {
+	for s.processNextWorkItem() {
+	}
+}
+
+func (s *StretchedNetworkPolicyController) processNextWorkItem() bool {
+	obj, quit := s.queue.Get()
+	if quit {
+		return false
+	}
+	defer s.queue.Done(obj)
+
+	if p, ok := obj.(*v1.Pod); !ok {
+		s.queue.Forget(obj)
+		klog.Errorf("Expected type 'pod' in work queue but got object", "object", obj)
+	} else if err := s.syncPodClassifierFlow(p); err == nil {
+		s.queue.Forget(p)
+	} else {
+		// Put the item back on the workqueue to handle any transient errors.
+		s.queue.AddRateLimited(p)
+		klog.ErrorS(err, "Error syncing podClassifierFlowUpdateItem, requeuing.", "podName", p.Name, "podNS", p.Namespace)
+	}
+	return true
+}
+
+// syncPodClassifierFlow will update labelToPods and podToLabel and update
+// PodClassifierFlow with its LabelIdentity, if available, or UnknownLabelIdentity.
+func (s *StretchedNetworkPolicyController) syncPodClassifierFlow(pod *v1.Pod) error {
+	podRef := s.getPodReference(pod)
+	normalizedLabel, err := s.getPodNormalizedLabels(pod)
+	if err != nil {
+		return err
+	}
+	if oldNormalizedLabel, ok := s.podToLabel[podRef]; ok {
+		s.deleteLabelToPod(oldNormalizedLabel, pod)
+	}
+	s.addLabelToPod(normalizedLabel, pod)
+	s.updatePodToLabel(normalizedLabel, pod)
+
+	if id, ok := s.labelIdentityCache[normalizedLabel]; ok {
+		return s.UpdatePodClassifierFlowWithLabelIdentity(pod, id)
+	}
+	return s.UpdatePodClassifierFlowWithLabelIdentity(pod, openflow.UnknownLabelIdentity)
+}
+
+// processPodUpdate handles Pod add events. It enqueues the Pod.
+func (s *StretchedNetworkPolicyController) processPodAdd(cur interface{}) {
+	curPod, curIsPod := cur.(*v1.Pod)
+	if !curIsPod {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Pod AddFunc received unexpected object", "curObj", cur)
+		return
+	}
+	s.enqueuePod(curPod)
```

Comment:
I think you don't need enqueuePod here since you already check the type.
```suggestion
	s.queue.Add(curPod)
```

Reply:
Good one. Done.